### PR TITLE
feat(185): Add api implementation

### PR DESCRIPTION
**Pull request by knguyenrise8**
# Pull Request

 - api implementation
 - pdf displays on processing page
 
 part 3
Screenshot 2024-09-25 at 9 15 06 AM

## Related Issues
185
## Additional Notes
[Add any additional context or notes that reviewers should know about.]

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.
- [ ] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: OCR/frontend/src/components/ExtractUploadFile.tsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ export const ExtractUploadFile: React.FC<ExtractUploadFileProps> = ({
       }
     }, 200);
   };
-
+  console.log(files)
```
Comment (zdeveloper) [inline OCR/frontend/src/components/ExtractUploadFile.tsx:86]: remove log

File: OCR/frontend/src/components/ImageAnnotator.tsx [hunk 2]
```diff
@@ -79,6 +79,7 @@ export const MultiImageAnnotator: FC<MultiImageAnnotatorProps> = ({ images, cate
                 ))}
             </div>
             <ImageAnnotator
+                id='image-annotator'
```
Comment (zdeveloper) [inline OCR/frontend/src/components/ImageAnnotator.tsx:82]: I thought this was not an attribute

File: OCR/frontend/api/api.ts [hunk 3]
```diff
@@ -1,18 +1,20 @@
-import { Page } from "../src/contexts/FilesContext";
-import { ImageToTextResponse } from "./types/types";
+import { ImageToTextArgs, ImageToTextResponse } from "./types/types";
 
-export const AddFormData = async (args: Page): Promise<ImageToTextResponse | null> => {
+export const AddFormData = async (args: ImageToTextArgs): Promise<ImageToTextResponse | null> => {
```
Comment (zdeveloper) [inline OCR/frontend/api/api.ts:3]: `AddFormData` is a bad name, a better name might be `imageToText`

File: OCR/frontend/src/pages/ExtractProcess.tsx [hunk 4]
```diff
@@ -60,9 +68,7 @@ const ExtractProcess = () => {
         file_image: templates[0].pages[0].sourceImage,
         results,
       }
-        
-      
-      
+              
       localStorage.setItem("extractedData", JSON.stringify(transformedResponses));
```
Comment (zdeveloper) [inline OCR/frontend/src/pages/ExtractProcess.tsx:72]: is this the correct place to save the returned data? I thought the review page was checking `submission` key

File: OCR/frontend/src/components/ExtractUploadFile.tsx (hunk 1 above)
Comment (knguyenrise8) [reply OCR/frontend/src/components/ExtractUploadFile.tsx:86]: 👍

File: OCR/frontend/src/components/ImageAnnotator.tsx (hunk 2 above)
Comment (knguyenrise8) [reply OCR/frontend/src/components/ImageAnnotator.tsx:82]: removed

File: OCR/frontend/api/api.ts [hunk 5]
```diff
@@ -1,18 +1,20 @@
...
-    form.append("", "");
-    console.log(args)
+    const form = new URLSearchParams({
+        source_image: sourceImage,
+        segmentation_template: templateImage,
+        labels: JSON.stringify(fieldNames),
+      });
+      
     try {
         const response = await fetch("http://localhost:8000/image_to_text/", {
```
Comment (zdeveloper) [inline OCR/frontend/api/api.ts:13]: extract `http://localhost:8000` into a environment variable, name it `API_URL`
https://vitejs.dev/guide/env-and-mode

File: OCR/frontend/src/pages/ReviewTemplate.tsx [hunk 6]
```diff
@@ -202,25 +217,34 @@ const ReviewTemplate: React.FC = () => {
         </div>
         <div className="width-50">
           <div
-            style={{
```
Comment (schreiaj) [inline OCR/frontend/src/pages/ReviewTemplate.tsx:205]: Removing this is what's causing the playwright test to break, I'd suggest removing the test `Document image scrollable` as it seems that is no longer a feature we have.

File: OCR/frontend/api/api.ts (hunk 5 above)
Comment (knguyenrise8) [reply OCR/frontend/api/api.ts:13]: updated

File: OCR/frontend/api/api.ts [hunk 7]
```diff
@@ -10,7 +10,7 @@ export const ImageToText = async (args: ImageToTextArgs): Promise<ImageToTextRes
       });
       
     try {
-        const response = await fetch("http://localhost:8000/image_to_text/", {
+        const response = await fetch(`${import.meta.env.APP_URL}image_to_text/`, {
```
Comment (zdeveloper) [inline OCR/frontend/api/api.ts:13]: change to API_URL

**Review by zdeveloper — CHANGES_REQUESTED** [review]

File: OCR/frontend/api/api.ts (hunk 7 above)
Comment (knguyenrise8) [reply OCR/frontend/api/api.ts:13]: yessir

**Comment by knguyenrise8** [comment]
> LGTM! thank you, really appreciate you incorporating all the feedback

no thanks for pairing and helping
